### PR TITLE
MUSARK-488

### PR DESCRIPTION
**Pull request by rituvesh**
Added date comparison login on common place
Added more unit test-cases to check the different dates.
Updated observation and control add logic with new logic.
Done the testing and impact testing.

### CODE REVIEW

File: src/util/index.js [hunk 1]
```diff
@@ -76,3 +76,11 @@ export const formatFloatToString = (number: number): string => {
 export const hasProp = (obj: any, prop: string): boolean => {
   return {}.hasOwnProperty.call(obj, prop)
 }
+
+export const isDateBiggerThanToday = (newDate: any): boolean => {
+  if (newDate) {
+    return (moment(newDate).isAfter()) ? true : false
```
Comment (jarlah) [inline src/util/index.js]: missing argument to isAfter. Or use moment(newDate).isSameOrBefore(todayDate) or similar. Its not readable to omit the argument.

Comment (kpmeen) [inline src/util/index.js]: I'm really curious to know why you feel it's necessary to use an Xor statement here?

Comment (rituvesh) [reply src/util/index.js]: Thanks for review. Removed.

**Review by jarlah — APPROVED** [review]

File: src/util/index.js (hunk 1 above)
Comment (rituvesh) [reply src/util/index.js]: ??? We have new code blow according to your suggestions.  
 const isAfterYear = moment(newDate).isAfter(today, 'year')
    const isAfterMonth = moment(newDate).isAfter(today, 'month')
    const isAfterDay = moment(newDate).isAfter(today, 'day')
     return isAfterDay || isAfterMonth || isAfterYear

Comment (jarlah) [reply src/util/index.js]: old comment ritu :)

Comment (rituvesh) [reply src/util/index.js]: :)

**Pull request merged**